packages/@ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -314,15 +351,19 @@ export class AngularCompilerPlugin implements Tapable {
           return;
         }
 
-        const factoryPath = discoveredLazyRoutes[lazyRouteKey]
-          .replace(/(\.d)?\.ts$/, '.ngfactory.js');
-        const factoryKey = `${lazyRouteModule}.ngfactory#${moduleName}NgFactory`;
+        const lazyRouteTSFile = discoveredLazyRoutes[lazyRouteKey];
+        let modulePath: string, moduleKey: string;
 
-        if (factoryKey in this._lazyRoutes) {
-          if (factoryPath === null) {
```

Comment:
You removed this part, which removed lazy routes that were deleted during a rebuild. Did you want to keep a check to remove lazy routes ?

Reply:
I checked earlier today and this actually doesn't do anything, as far as I could tell. When a lazy route goes away the discovered routes for this module is an empty object, so there's nothing to remove. There's a TODO just above to actually add this functionality.